### ISSUE TITLE
crash using stdout after GLFW.terminate "<stdout>: commitBuffer: invalid argument (Bad file descriptor)"

### ISSUE THREAD

**Issue by plredmond** [bsl/GLFW-b#54]
this program crashes when the last line, contaning `print`, is evaluated
```haskell
import qualified Graphics.UI.GLFW as GLFW

main :: IO ()
main = do
    GLFW.init >>= print
    print "inited"

    GLFW.terminate
    print "terminated"
```
the program output looks like this:
```
True
"inited"
Main2: <stdout>: commitBuffer: invalid argument (Bad file descriptor)
```

---

a similar program, written in C, doesn't crash
```c
#include <GLFW/glfw3.h>
#include <stdio.h>

int main(void)
{
    printf("%d\n", glfwInit());
    printf("inited\n");

    glfwTerminate();
    printf("terminated\n");
}
```
the c program output looks like this:
```
1
inited
terminated
```

---

i've done some preliminary messing around and found

* if you write to `Sysetm.IO.stdout` using `System.IO.hPrint`, it crashes with the same message
* if you examine `System.IO.stdout` using `System.IO.{hIsOpen,hIsClosed,hIsReadable,hIsWritable,hIsSeekable` before ande after calling `GLFW.terminate`, the descriptor reports that is is okay

---

my versions, etc..

* system
  * ArchLinux, kernel 4.0.6-1-ARCH
* haskell
  * GHC 7.8.4-1
  *  cabal-install version 1.22.2.0 using version 1.22.2.0 of the Cabal library
    * GLFW-b-1.4.7.2
    * bindings-GLFW-3.1.1.3
* C code
  * gcc (GCC) 5.1.0
  * glfw 3.1.1-1 (installed on archlinux via pacman)

**Comment by plredmond** [comment]
i'd be happy to help with this .. i dug into your code a bit ([terminate](https://github.com/bsl/GLFW-b/blob/master/Graphics/UI/GLFW.hs#L299)] and just saw a thin layer over the `bindings-GLFW` function, so i'm not really sure where to poke further

**Comment by schell** [comment]
By all means - any help you can give is much appreciated! :)

**Comment by plredmond** [comment]
my suspicion is is that something in the haskell ffi has to reassign stdin/stdout/stderr file descriptors to the c code before each c-invocation, and then restore them afterward.. somehow this restoration is disrupted and causing the crash .. i'll poke around and get back to you!

**Comment by schell** [comment]
That sounds great. Thank you!

**Comment by JavierJF** [comment]
Thanks for the report@plredmond, this has been fixed by commit JavierJF/bindings-GLFW@db8970a713a032c1e0383a0214f7f47272718bb3
